packages/grafana-data/src/datetime/datemath.test.ts

```diff
@@ -52,6 +52,22 @@ describe('DateMath', () => {
     expect(startOfDay).toBe(expected.getTime());
   });
 
+  describe('time zones', () => {
+    it('parse should take timezones into account', () => {
+      const dateToParse = new Date(1590674400000); //  Thursday, May 28, 2020 2:00:00 PM
+      const timezone = 'Europe/Istanbul'; // +3 UTC
+
+      const parsedDate = dateMath.parse(dateToParse, undefined, timezone)!;
+      expect(parsedDate.toString()).toBe('Thu May 28 2020 17:00:00 GMT+0300');
```

Comment:
I'm not sure I agree that this code behaves correctly. But it can be my lack of understanding of what the `parse` function should do. The `Thursday, May 28, 2020 2:00:00 UTC` and `Thu May 28 2020 17:00:00 GMT+0300` is the same date. You are just displaying it in different time zones.

So there are no need to parse the date, in my opinion. If you want to display the date in the `Europe/Istanbul` you can just use the `dateToParse` and format in the timezone you would like to have.

If it would have been a string, that you wan to parse, without the time zone specified in it e.g. `Thursday, May 28, 2020 2:00:00` then this test would make sense because you don't know in what time zone the date is defined in. But when you create a date using the epoch it is defined in UTC.

Does this make sense?